ntoskrnl/io/pnpmgr/pnpmgr.c

```diff
@@ -1760,6 +1765,81 @@ IopGetParentIdPrefix(PDEVICE_NODE DeviceNode,
     return Status;
 }
 
+static
+ULONG
```

Comment:
Yes

Reply:
After looking in more detail at how this function is currently used in the code, it may be better to return BOOLEAN instead (FALSE: invalid IDs; TRUE: valid IDs).
Because the `TotalLength` thing is actually used only to enumerate the multi-strings in the code, and the value returned by this function IopValidateID is not really used.

Reply:
At first I thought to replace this code:
```
      Ptr = (PWSTR)IoStatusBlock.Information;
      DPRINT("Hardware IDs:\n");
      while (*Ptr)
      {
         DPRINT("  %S\n", Ptr);
         Length = (ULONG)wcslen(Ptr) + 1;

         Ptr += Length;
         TotalLength += Length;
      }
      DPRINT("TotalLength: %hu\n", TotalLength);
      DPRINT("\n");
```
But now, since I did not do it, yes, can use BOOLEAN.